common/src/main/java/com/wynntils/features/utilities/TranscribeMessagesFeature.java

```diff
@@ -0,0 +1,173 @@
+
+    @Persisted
+    public final Config<TranscribeCondition> transcribeCondition = new Config<>(TranscribeCondition.ALWAYS);
+
+    @Persisted
+    public final Config<Boolean> coloredTranscriptions = new Config<>(true);
+
+    @Persisted
+    public final Config<ColorChatFormatting> gavellianColor = new Config<>(ColorChatFormatting.LIGHT_PURPLE);
+
+    @Persisted
+    public final Config<ColorChatFormatting> wynnicColor = new Config<>(ColorChatFormatting.DARK_GREEN);
+
+    private static final Pattern END_OF_HEADER_PATTERN = Pattern.compile(".*[\\]:]\\s?");
+    private static final Pattern WYNNIC_NUMBER_PATTERN = Pattern.compile("[⑴-⑿]+");
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onChat(ChatMessageReceivedEvent event) {
+        if (!transcribeChat.get()) return;
+        if (!Models.WynnAlphabet.hasWynnicOrGavellian(event.getStyledText().getString())) return;
+
+        boolean transcribeWynnic = Models.WynnAlphabet.shouldTranscribe(transcribeCondition.get(), WynnAlphabet.WYNNIC);
+        boolean transcribeGavellian =
+                Models.WynnAlphabet.shouldTranscribe(transcribeCondition.get(), WynnAlphabet.GAVELLIAN);
+
+        if (!transcribeWynnic && !transcribeGavellian) return;
+
+        StyledText styledText = event.getStyledText();
+
+        StyledText modified = getStyledTextWithTranscription(styledText, transcribeWynnic, transcribeGavellian, false);
+
+        if (styledText.getString().equalsIgnoreCase(modified.getString())) return;
+
+        event.setMessage(modified.getComponent());
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onNpcDialogue(NpcDialogEvent event) {
+        if (!transcribeNpcs.get()) return;
+        if (!Models.WynnAlphabet.hasWynnicOrGavellian(event.getChatMessage().toString())) return;
+
+        boolean transcribeWynnic = Models.WynnAlphabet.shouldTranscribe(transcribeCondition.get(), WynnAlphabet.WYNNIC);
+        boolean transcribeGavellian =
+                Models.WynnAlphabet.shouldTranscribe(transcribeCondition.get(), WynnAlphabet.GAVELLIAN);
+
+        if (!transcribeWynnic && !transcribeGavellian) return;
+
+        event.setCanceled(true);
+
+        if (!event.getChatMessage().isEmpty()) {
+            List<Component> transcriptedComponents = event.getChatMessage().stream()
+                    .map(styledText -> getStyledTextWithTranscription(
+                            StyledText.fromComponent(styledText), transcribeWynnic, transcribeGavellian, true))
+                    .map(s -> ((Component) s.getComponent()))
+                    .toList();
+
+            Managers.TickScheduler.scheduleNextTick(() -> {
+                NpcDialogEvent transcriptedEvent = new WynnTranscriptedNpcDialogEvent(
+                        transcriptedComponents, event.getType(), event.isProtected());
+                WynntilsMod.postEvent(transcriptedEvent);
+            });
+        } else {
+            NpcDialogEvent transcriptedEvent =
```

Comment:
Can this ever happen? If there is no message, then Models.WynnAlphabet.hasWynnicOrGavellian can hardly return true, can it?

Reply:
Might not be, I got this from the onNpcDialogue in the translation feature